### PR TITLE
Consolidate Calamari.Scripting

### PR DESCRIPTION
**Pull request by mik-ky**
Moving Calamari.Scripting and its test in [Sashimi repo](https://github.com/OctopusDeploy/Sashimi/tree/main/source) as part of dependency management work tranche 2.

**Changes**
* Pipeline to build Calamari.Scripting and tests
* Rest of the flavours referencing Calamari.Common are now project references.

### CODE REVIEW

File: source/Calamari.AzureScripting/Calamari.AzureScripting.csproj [hunk 1]
```diff
@@ -8,21 +8,25 @@
         <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
         <RuntimeIdentifiers>win-x64;linux-x64;osx-x64;linux-arm;linux-arm64</RuntimeIdentifiers>
         <IsPackable>true</IsPackable>
+        <ValidateExecutableReferencesMatchSelfContained>false</ValidateExecutableReferencesMatchSelfContained>
```
Comment (YihaoOct) [inline source/Calamari.AzureScripting/Calamari.AzureScripting.csproj:11]: This is to make this project able to include another project that outputs a self-contained exe, am I understanding it correctly?

File: source/Calamari.AzureCloudService/Calamari.AzureCloudService.csproj [hunk 2]
```diff
@@ -26,6 +24,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Calamari.AzureScripting\Calamari.AzureScripting.csproj" />
+    <ProjectReference Include="..\Calamari.Common\Calamari.Common.csproj" />
+    <ProjectReference Include="..\Calamari.Scripting\Calamari.Scripting.csproj" />
```
Comment (YihaoOct) [inline source/Calamari.AzureCloudService/Calamari.AzureCloudService.csproj:28]: Not sure if these are needed, I think they should have been transitively included from `Calamari.AzureScripting` now. But I'm happy either way

**Review by YihaoOct — APPROVED** [review]
LGTM, I have left some suggestions~

File: source/Calamari.AzureScripting/Calamari.AzureScripting.csproj (hunk 1 above)
Comment (mik-ky) [reply source/Calamari.AzureScripting/Calamari.AzureScripting.csproj:11]: Yup, that's correct.

File: build/Build.cs [hunk 3]
```diff
@@ -213,85 +213,99 @@ public Build()
...
-                                           .SetRuntime(packageToBuild.Architecture)
-                                           .SetOutput(outputDirectory)
-                                     );
-                        SignDirectory(outputDirectory);
-
-                        File.Copy(RootDirectory / "global.json", outputDirectory / "global.json");
-                    }
-
-                    foreach (var flavour in calamariFlavours)
+        void PublishCalamariProjects(IEnumerable<Project> projects)
```
[the remark this answers is not in the document]
Comment (mik-ky) [reply build/Build.cs:230]: Let's keep this for now. I'm more than happy to clean up the build afterwards :)

**Pull request merged**